lib/eph/checkpointInfo.ts

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+import * as debugModule from "debug";
+const debug = debugModule("azure:event-hubs:eph:checkpointInfo");
+
+/**
+ * Describes the checkoint information.
+ * @interface CheckpointInfo
+ */
+export interface CheckpointInfo {
+  /**
+   * @property {string} partitionId The EventHub partition id.
+   */
+  partitionId: string;
+  /**
+   * @property {string} owner The name of the owner checking in this event (usually the eph host name).
+   */
+  owner: string;
+  /**
+   * @property {string} token The token associated with the lease.
+   */
+  token: string;
+  /**
+   * @property {string} epoch The epoch value associated with the receiver that is checking in this
+   * event.
+   */
+  epoch: number;
+  /**
+   * @property {string} offset The offset of the event to be checked in.
+   */
+  offset?: string;
+  /**
+   * @property {string} sequenceNumber The sequence number of the event to be checked in.
+   */
+  sequenceNumber: number;
+}
+
+/**
+ * Describes the .net sdk's way of checkpointing information. If someone has previously used .net
+ * sdk then we should be able to use that information.
+ */
+export interface CheckpointInfoUpperCase {
+  PartitionId: string;
+  Owner: string;
+  Token: string;
+  Epoch: number;
+  Offset?: string;
+  SequenceNumber: number;
+}
+
```

Comment:
I was wondering if someone uses .net sdk EPH implementation and then uses the node sdk EPH implementation then the node one should not start from -1 (i.e. the beginning) because it only understands camel cased JSON object keys. In js object keys are case sensitive. Hence thought of writing to the blob in the same format as the .net sdk. However our checkpointinfo interface follows the standard js naming conventions.

I see your concerns as well. Will remove this and probably only have support for deserializing the blob content for both the cases. However when we write the checkpoint info to blob we will follow the standard camel casing as used by other (go, python  sdks),